src/jdk.jdwp.agent/share/native/libjdwp/utf_util.c

```diff
@@ -333,7 +336,7 @@ static WCHAR* getWideString(UINT codePage, char* str, int len, int *pwlen) {
     wlen = MultiByteToWideChar(codePage, 0, str, len, NULL, 0);
     *pwlen = wlen;
     if (wlen <= 0) {
-        UTF_ERROR(("Can't get WIDE string length"));
+        UTF_WARNING(("Can't get WIDE string length"));
```

Comment:
If you are changing this to a warning then it should probably be limited to the NO_UNICODE_TRANSLATION case, I think you still want to abort if it fails due to invalid flags or parameters or buffer issue as that would a JDK (or Windows) bug.

Reply:
If you are going to do that, then you should track down all uses of this code (direct and indirect) to see if this error is ever acceptable and is handled properly. It seems that if the unicode string is bad, we should be exiting. I see this code being used indirectly from setAgentPropertyValue(), which only seems to be used for sun.jdwp.listenerAddress. Do we want the raw unicode bytes to be used in this case if they cannot be converted? It's also used from printLastError(), which gets the unicode bytes from GetLastError(). Should't they always be valid?

Reply:
> If you are going to do that, then you should track down all uses of this code (direct and indirect) to see if this error is ever acceptable and is handled properly. It seems that if the unicode string is bad, we should be exiting. I see this code being used indirectly from setAgentPropertyValue(), which only seems to be used for sun.jdwp.listenerAddress. Do we want the raw unicode bytes to be used in this case if they cannot be converted? It's also used from printLastError(), which gets the unicode bytes from GetLastError(). Should't they always be valid?

I agree it complicates things and it would require looking at all usages. My comment was just to say that if it is changed to emit a warning then it would only make sense to do so in limited cases (the NO_UNICODE_TRANSLATION mostly). It may be simpler to not change the code of course.